vsintegration/tests/Salsa/VsMocks.fs

```diff
@@ -1647,10 +1647,17 @@ module internal VsActual =
     let vsInstallDir =
         // use the environment variable to find the VS installdir
         let vsvar = 
-            let var = Environment.GetEnvironmentVariable("VS150COMNTOOLS")
-            if String.IsNullOrEmpty var then Environment.GetEnvironmentVariable("VSAPPIDDIR") 
-            else var
-        if String.IsNullOrEmpty vsvar then failwith "VS150COMNTOOLS and VSAPPIDDIR environment variables not found."
+            let var =
```

Comment:
You're better off using `vswhere` - it's always installed to a fixed location and doesn't depend on environment variables:

```
> %ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe -latest -format value -property installationPath
C:\Program Files (x86)\Microsoft Visual Studio\2017\Professional
```

---

From the Microsoft/vswhere readme:

> vswhere is included with the installer as of Visual Studio 2017 version 15.2 and later, and can be found at the following location: `%ProgramFiles(x86)%\Microsoft Visual Studio\Installer\vswhere.exe`

Reply:
I think I agree ... maybe.  But I would prefer it if these tests just crawled away and died.  They are unreliable, opaque and a lot of trouble.